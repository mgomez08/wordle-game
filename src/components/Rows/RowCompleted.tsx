import checkLetter from "../../helpers/checkLetter";
import { Box } from "../Box/Box";

import style from "./row.module.scss";
interface RowCompletedProps {
  word: string;
  solution: string;
}

export const RowCompleted = ({ word, solution }: RowCompletedProps) => {
  const ROWS: number = 5;
  const arrayRows = Array.from(Array(ROWS));
  return (
    <div className={style.row}>
      {arrayRows.map((_, index) => (
        <Box
          key={index}
          value={word[index]}
          status={checkLetter(word[index], index, solution)}
        />
      ))}
    </div>
  );
};
