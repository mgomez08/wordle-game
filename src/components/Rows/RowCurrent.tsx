import { Box } from "../Box/Box";
import styles from "./row.module.scss";

interface RowCurrentProps {
  word: string;
}

export const RowCurrent = ({ word }: RowCurrentProps) => {
  const wordArray = word.split("");
  return (
    <div className={styles.row}>
      {wordArray.map((letter, index) => (
        <Box key={index} value={letter} status="edit" />
      ))}
      {Array.from(Array(5 - wordArray.length)).map((letter, index) => (
        <Box key={index} value={""} status="edit" />
      ))}
    </div>
  );
};
