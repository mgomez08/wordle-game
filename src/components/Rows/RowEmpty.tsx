import { Box } from "../Box/Box";
import styles from "./row.module.scss";

export const RowEmpty = () => {
  const ROWS: number = 5;
  const arrayRows = Array.from(Array(ROWS));
  return (
    <div className={styles.row}>
      {arrayRows.map((_, index) => (
        <Box key={index} value="" status="empty" />
      ))}
    </div>
  );
};
