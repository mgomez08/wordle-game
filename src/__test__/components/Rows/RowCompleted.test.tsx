import { mount } from "enzyme";
import { RowCompleted } from "../../../components/Rows/RowCompleted";

describe("Tests for <RowCompleted />", () => {
  const rowcompleted = mount(<RowCompleted word="HELLO" solution="HELLO" />);

  test("Render RowCompleted component", () => {
    expect(rowcompleted.length).toEqual(1);
  });

  test("RowCompleted component should have correct number of children (boxes)", () => {
    expect(rowcompleted.find("Box").length).toEqual(5);
  });

  test("First Box children should have class (.correct) in case the word and the solution are the same", () => {
    expect(rowcompleted.find("Box").first().find(".correct").length).toEqual(1);
  });

  test("First Box children should have class (.absent) in case the word and the solution are different", () => {
    const rowcompleted = mount(<RowCompleted word="HELLO" solution="BATTY" />);
    expect(rowcompleted.find("Box").first().find(".absent").length).toEqual(1);
  });

  test("First Box children should have class (.present) in case the word contains a letter of the solution", () => {
    const rowcompleted = mount(<RowCompleted word="HAPPY" solution="FLESH" />);
    expect(rowcompleted.find("Box").first().find(".present").length).toEqual(1);
  });

  test("RowCompleted component with Snapshot", () => {
    expect(rowcompleted).toMatchSnapshot();
  });
});
