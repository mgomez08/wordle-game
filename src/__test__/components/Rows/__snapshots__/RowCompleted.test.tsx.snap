// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <RowCompleted /> RowCompleted component with Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="correct"
  >
    H
  </div>
  <div
    className="correct"
  >
    E
  </div>
  <div
    className="correct"
  >
    L
  </div>
  <div
    className="correct"
  >
    L
  </div>
  <div
    className="correct"
  >
    O
  </div>
</div>
`;
